float64 radius
float64 distance
